File: client/src/types/knowledge.ts
import type { UserBrief } from './user'

export type DocumentStatus = 'PROCESSING' | 'READY' | 'FAILED'
export type DocumentFileType = 'TXT' | 'MARKDOWN' | 'PDF'

export interface KnowledgeDocument {
  id: number
  title: string
  filename: string | null
  file_type: DocumentFileType
  status: DocumentStatus
  error_message: string | null
  chunk_count: number
  embedded_chunk_count: number
  uploaded_by: UserBrief | null
  created_at: string
  updated_at: string
}

export interface KnowledgeDocumentDetail extends KnowledgeDocument {
  content: string
}

export interface KnowledgeSearchResult {
  chunk_id: number
  document_id: number
  document_title: string
  section: string | null
  chunk_text: string
  score: number
}

export interface KnowledgeSearchResponse {
  mode: 'semantic' | 'keyword'
  results: KnowledgeSearchResult[]
}

File: server/app/schemas/knowledge.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict

from app.models import DocumentFileType, DocumentStatus
from app.schemas.user import UserBrief


class DocumentOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    title: str
    filename: str | None
    file_type: DocumentFileType
    status: DocumentStatus
    error_message: str | None
    chunk_count: int
    embedded_chunk_count: int
    uploaded_by: UserBrief | None
    created_at: datetime
    updated_at: datetime


class DocumentDetail(DocumentOut):
    content: str


class SearchResultOut(BaseModel):
    chunk_id: int
    document_id: int
    document_title: str
    section: str | None
    chunk_text: str
    score: float


class SearchResponse(BaseModel):
    mode: Literal["semantic", "keyword"]
    results: list[SearchResultOut]
